### PR TITLE
Add stricter version of `parseFloat()`: `stringToNumber()`

### PR DESCRIPTION
**Pull request by brendandagys**
Included tests.

Added `test-watch` script in `package.json`.

### CODE REVIEW

File: src/misc/stringToNumber.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** Returns a numeric representation of an input string.
```
Comment (jmarks-lws) [inline src/misc/stringToNumber.ts:1]: @brendandagys This description should start on the next line to maintain consistency with the rest of the project.

File: src/misc/stringToNumber.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+ * The string represents one or more digits, with exactly one
+ * optional decimal place, at any location in the string.
+ * This may return NaN.
+ * @param str String that represents a number, optionally with one decimal place.
+ * */
+export const stringToNumber = (str: string): number => (
+  /^((\d+(\.\d*)?)|\.\d+)$/.test(str) ? parseFloat(str) : NaN
+);
```
Comment (jmarks-lws) [inline src/misc/stringToNumber.ts:9]: @brendandagys I believe having one space at the end of the file is part of the linting rules. Did you apply eslint with the rules used by the project's `.eslintrc` to your submission?

File: src/misc/stringToNumber.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/** Returns a numeric representation of an input string.
+ * The string represents one or more digits, with exactly one
+ * optional decimal place, at any location in the string.
+ * This may return NaN.
+ * @param str String that represents a number, optionally with one decimal place.
+ * */
```
Comment (jmarks-lws) [inline src/misc/stringToNumber.ts:6]: Let's try:
```
/** 
 * Returns a numeric representation of an input string or `NaN` if the provided string value is not valid.
 * The string should consist one or more digits, with zero or one decimal point (period) within the string.
 * If the string does not match this pattern, this function will return `NaN`.
 * @param str String that should represent a number, with one optional decimal point.
 * */
 ```

File: src/misc/stringToNumber.ts (hunk 1 above)
Comment (brendandagys) [reply src/misc/stringToNumber.ts]: I have replaced the JSDoc with the snippet from your other comment, and now the description starts on the next line.

File: src/misc/stringToNumber.ts (hunk 2 above)
Comment (brendandagys) [reply src/misc/stringToNumber.ts:9]: I read a while ago that having an empty line at the end of a document is a good practice as you suggest, and I strive to adhere to that. I double-checked the code that I checked in, and there is in fact a blank line at the end of the file.

I believe that GitHub is truncating it. Looking at other files on GitHub in your library show the omission of it as well, so I believe this is a UI quirk.

**Pull request merged**